packages/medusa/src/types/common.ts

```diff
@@ -164,6 +168,28 @@ export class StringComparisonOperator {
   lte?: string
 }
 
+export class StringSearchOperator {
```

Comment:
thought(non-blocking): Are these necessary? Wouldn't it suffice to use our existing text search mechanism?

Reply:
I think it is nice to have but maybe not a must, something we can discuss but otherwise I ve written a comment to move the prepare method to an util as part of the buildQuery utils.

Reply:
Yeah, think it is a nice-to-have, but I am fine keeping it in. I like the verbosity of this approach compared to alternatives. And this is consistent with the `in` and nin` operators introduced in the products module. 

@srindom – I would like you to sign off on this too :)